src/frequenz/sdk/actor/_power_managing/_matryoshka.py

```diff
@@ -217,12 +221,23 @@ def _calc_target_power(
             if upper_bound < lower_bound:
                 break
             if next_proposal.preferred_power:
-                target_power = _clamp_to_bounds(
+                match _clamp_to_bounds(
                     next_proposal.preferred_power,
                     lower_bound,
                     upper_bound,
                     exclusion_bounds,
-                )
+                ):
+                    case (None, power) | (power, None) if power:
+                        target_power = power
+                    case (power_low, power_high) if power_low and power_high:
```

Comment:
:exploding_head: I didn't remember the `if` syntax.

Reply:
yes, the whole thing is very similar to the `pcase` in lisp, and the `match` from rust.

Reply:
And D's [template constraints](https://dlang.org/spec/template.html#template_constraints) :laughing:

Reply:
wow, wouldn't have guessed D also has functional pattern matching.